Job build error

```
[root@master ~]# jcli job search publish |grep demo
      "Name": "test_publish_demo"
      "Name": "ops_auto_dev_go_publish_demo"
      "Name": "ops_auto_pre_go_publish_demo"
      "Name": "ops_all_pkg_java_publish_demo"
      "Name": "ops_auto_dev_go_publish_demo1"
[root@master ~]# jcli job build -n test_publish_demo
? Are you sure to build job test_publish_demo Yes
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6bf987]

goroutine 1 [running]:
github.com/linuxsuren/jenkins-cli/client.(*JenkinsCore).CrumbHandle(0xc00018fcf0, 0xc0001c6000, 0xd, 0xc000015e80)
 /Users/mac/go/src/github.com/linuxsuren/jenkins-cli/client/core.go:55 +0x47
github.com/linuxsuren/jenkins-cli/client.(*JobClient).Build(0xc00018fcf0, 0x7ffd543466c8, 0x11, 0x1, 0x1)
 /Users/mac/go/src/github.com/linuxsuren/jenkins-cli/client/job.go:72 +0x31b
github.com/linuxsuren/jenkins-cli/app/cmd.glob..func9(0xbdac60, 0xc00016c160, 0x0, 0x2)
 /Users/mac/go/src/github.com/linuxsuren/jenkins-cli/app/cmd/job_build.go:43 +0x235
github.com/spf13/cobra.(*Command).execute(0xbdac60, 0xc00016c100, 0x2, 0x2, 0xbdac60, 0xc00016c100)
 /Users/mac/go/src/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xbdc1c0, 0x869843, 0x1, 0x0)
 /Users/mac/go/src/github.com/spf13/cobra/command.go:852 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
 /Users/mac/go/src/github.com/spf13/cobra/command.go:800
github.com/linuxsuren/jenkins-cli/app/cmd.Execute()
 /Users/mac/go/src/github.com/linuxsuren/jenkins-cli/app/cmd/root.go:40 +0x32
main.main()
 /Users/mac/go/src/github.com/linuxsuren/jenkins-cli/main.go:8 +0x20
[root@master ~]# jcli job build -n test_publish_demo -b
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6bf987]

goroutine 1 [running]:
github.com/linuxsuren/jenkins-cli/client.(*JenkinsCore).CrumbHandle(0xc00006fcf0, 0xc00014b600, 0xd, 0xc000015e80)
 /Users/m
```

--- a/app/cmd/job_build.go
+++ b/app/cmd/job_build.go
@@ -46,7 +46,6 @@ var jobBuildCmd = &cobra.Command{
 		paramDefs := []client.ParameterDefinition{}
 		hasParam := false
 		if job, err := jclient.GetJob(jobOption.Name); err == nil {
-			fmt.Println(job.Property)
 			if len(job.Property) != 0 {
 				for _, pro := range job.Property {
 					if len(pro.ParameterDefinitions) == 0 {

--- a/client/core.go
+++ b/client/core.go
@@ -51,7 +51,8 @@ func (j *JenkinsCore) AuthHandle(request *http.Request) {
 }
 
 func (j *JenkinsCore) CrumbHandle(request *http.Request) error {
-	if c, err := j.GetCrumb(); err == nil {
+	if c, err := j.GetCrumb(); err == nil && c != nil {
+		// cannot get the crumb could be a noraml situation
 		j.CrumbRequestField = c.CrumbRequestField
 		j.Crumb = c.Crumb
 		request.Header.Add(j.CrumbRequestField, j.Crumb)
@@ -80,7 +81,10 @@ func (j *JenkinsCore) GetCrumb() (*JenkinsCrumb, error) {
 		if data, err := ioutil.ReadAll(response.Body); err == nil {
 			if response.StatusCode == 200 {
 				json.Unmarshal(data, &crumbIssuer)
+			} else if response.StatusCode == 404 {
+				return nil, err
 			} else {
+				log.Printf("Unexpected status code: %d.", response.StatusCode)
 				return nil, err
 			}
 		} else {

